src/webserver/WsBase.h

```diff
@@ -245,48 +245,31 @@ class WsSession : public WsBase,
     }
 
     void
-    handle_request(std::string const&& msg, boost::asio::yield_context& yc)
+    handle_request(
+        boost::json::object&& request,
+        const boost::json::value& id,
+        boost::asio::yield_context& yc)
     {
         auto ip = derived().ip();
         if (!ip)
             return;
 
         boost::json::object response = {};
-        auto sendError = [this](auto error, boost::json::value id) {
+        auto sendError = [this, &request, id](auto error) {
             auto e = RPC::make_error(error);
-
-            if (!id.is_null())
-                e["id"] = id;
-
+            e["id"] = id;
```

Comment:
What happens when `id` is null?

Reply:
if `id` is left off the request, it should also be left off the response

Reply:
Fixed.